cli/src/main/java/com/liferay/blade/cli/util/ProductKeyUtil.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.blade.cli.util;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Drew Brokke
+ */
+public class ProductKeyUtil {
+
+	public static final Comparator<String> comparator = ProductKeyUtil::compare;
+	public static final Pattern productKeyCommercePattern = Pattern.compile(
+		"^(?<product>commerce)-(?<major>[1-9]\\.\\d\\.\\d)(?:-(?<minor>[1-9]\\.\\d))?$");
+	public static final Pattern productKeyDXPNonquarterlyPattern = Pattern.compile(
+		"^(?<product>dxp)-(?<major>[1-9]\\.\\d)-(?<minor>(?:de|ep|fp|ga|sp|u)\\d+)$");
+	public static final Pattern productKeyDXPQuarterlyPattern = Pattern.compile(
+		"^(?<product>dxp)-(?<major>2\\d{3})\\.(?<minor>q[1234])\\.(?<micro>\\d+)$");
+	public static final Pattern productKeyPortalPattern = Pattern.compile(
+		"^(?<product>portal)-(?<major>[1-9]\\.\\d)-(?<minor>ga\\d+)$");
+	public static final List<String> products = Collections.unmodifiableList(
+		Arrays.asList("commerce", "portal", "dxp"));
```

Comment:
Remind me again why this can't be sorted? `"commerce", "dxp", "portal"`?

Reply:
Now I think this needs to change because it was not immediately obvious...

Reply:
I just renamed the field to `_orderedProducts` so it's at least known that they are intentionally ordered.